enterprise/server/remote_execution/copy_on_write/copy_on_write.go

```diff
@@ -506,6 +506,8 @@ func (s *COWStore) initDirtyChunk(offset int64, size int64) (ogChunk *Mmap, newC
 	ogChunk = s.chunks[offset]
 	chunkSource := snaputil.ChunkSourceHole
 	if ogChunk != nil {
+		ogChunk.mu.RLock()
+		defer ogChunk.mu.RUnlock()
```

Comment:
maybe simpler to unlock after assigning to chunkSource?

Reply:
Done

Reply:
Actually ended up changing the implementation